packages/appeals-service-api/src/controllers/appeals.js

```diff
@@ -18,199 +17,71 @@ module.exports = {
     appeal.updatedAt = now;
 
     logger.debug(`Creating appeal ${appeal.id} ...`);
-    try {
-      await mongodb
-        .get()
-        .collection('appeals')
-        .insertOne({ _id: appeal.id, uuid: appeal.id, appeal });
 
+    const document = await insertAppeal(appeal);
+
+    if (document.result.ok) {
       logger.debug(`Appeal ${appeal.id} created`);
       res.status(201).send(appeal);
-    } catch (err) {
-      logger.error(`Problem creating an appeal ${appeal.id}\n${err}`);
-      res.status(500).send(`Problem creating an appeal`);
+      return;
     }
+
+    logger.error(`Problem while ${appeal.id} created`);
+    res.status(500).send(appeal);
   },
 
   async getAppeal(req, res) {
     const idParam = req.params.id;
+
     logger.debug(`Retrieving appeal ${idParam} ...`);
     try {
-      await mongodb
-        .get()
-        .collection('appeals')
-        .findOne({ _id: idParam })
-        .then((doc) => {
-          logger.debug(`Appeal ${idParam} retrieved`);
-          res.status(200).send(doc.appeal);
-        })
-        .catch((err) => {
-          logger.warn(`Could not find appeal ${idParam}\n${err}`);
-          res.status(404).send(null);
-        });
-    } catch (err) {
-      logger.error(`Problem retrieving appeal ${idParam}\n${err}`);
-      res.status(500).send(`Problem finding appeal ${idParam}`);
+      const document = await getAppeal(idParam);
```

Comment:
that's confusing - fn name of `getAppeal` shadows the fn on line 33

maybe

```
const { getAppeal: getAppealFromAppealApiService, updateAppeal, insertAppeal } = require('../services/appeal.service');

...

 const document = await getAppealFromAppealApiService(idParam);
```

idk - naming is probably wrong, but on initial read it looks like you're calling this again recursively